### PR TITLE
Disable xdebug on cli

### PR DESCRIPTION
**Pull request by Toflar**
This will disable xdebug automatically when it's enabled by default for increased performance.
In 99.999999% you never want xdebug to be enabled when running commands on our CLI interfaces. If you want to debug something, you can still enable it setting the respective env var.

### CODE REVIEW

**Comment by aschempp** [comment]
why shouldn't that be a default configuration in your PHP CLI setup?

**Comment by Toflar** [comment]
I'd say it's pretty normal to have xdebug enabled as an extension locally and disablilng it on the fly is pretty hard (which is why there's `composer/xdebug-handler`).

**Comment by m-vo** [comment]
I don't like this (and don't like it with composer/phpstan/... neither). It's really unintuitive if you start debugging  and suddenly it's not working anymore. Having xdebug enabled on CLI is a bad practice imo. You start the process -with `-dxdebug.remote_enable=1` if you want to debug.

**Comment by leofeyer** [comment]
I agree with @m-vo. @ausi WDYT?

**Comment by Toflar** [comment]
I disagree. We ship a Managed Edition here so we should aim for the majority (= users that do not have a clue about which extensions are installed), not the minority (us knowing which ones to disable or enable).
And as a matter of fact, there are thousands of server setups which have xdebug installed.
Having xdebug installed **already** poses the performance issue. It doesn't matter whether tracing/profiling is enabled or not. So your `-d xdebug.remote_enable=1` won't fix a thing.  That's also the reason why Composer disables it automatically. It would take ages to resolve the dependencies even though profiling is not even enabled.

File: manager-bundle/bin/contao-api [hunk 1]
```diff
@@ -28,5 +29,10 @@ if (file_exists(__DIR__.'/../../../../autoload.php')) {
 /** @var Composer\Autoload\ClassLoader $loader */
 $loader = require $projectDir.'/vendor/autoload.php';
 
+// Automatically disable xdebug if enabled. You can enable it running "CONTAO_API_ALLOW_XDEBUG=1 vendor/bin/contao-api"
+$xdebug = new XdebugHandler('CONTAO_API');
+$xdebug->check();
+unset($xdebug);
```
Comment (leofeyer) [inline manager-bundle/bin/contao-api:35]:
```suggestion
(new XdebugHandler('CONTAO_API'))->check();
```

Comment (Toflar) [reply manager-bundle/bin/contao-api:35]: It's copied from the docs. Not sure but maybe having memory freed explicitly there has a reason ;)

Comment (leofeyer) [reply manager-bundle/bin/contao-api:35]: If you do not set the variable, nothing remains in memory that needed to be freed IMHO. 🤷🏻‍♂️

**Comment by aschempp** [comment]
> I disagree. We ship a Managed Edition here so we should aim for the majority (= users that do not have a clue about which extensions are installed), not the minority (us knowing which ones to disable or enable).
> And as a matter of fact, there are thousands of server setups which have xdebug installed.

If that would be true, shouldn't we force-disable Xdebug on the web entry point as well?
(not saying that I'd agree 😆 )

**Comment by Toflar** [comment]
If we could, we should. But we can't because we cannot start a subprocess of our own in the web process ;)

**Comment by ausi** [comment]
> @ausi WDYT?

I honestly don’t know... should we set this up for discussion?

**Comment by m-vo** [comment]
I still think that's the problem of Xdebug/server config. You would have this problem with every single application. Imo Xdebug should never be installed on a production machine.

If the performance penalty from just having it enabled is a problem when developing, you could for example also add a bash alias for your php process and enable the extension on the fly: https://stackoverflow.com/a/61018119/3146819

**Comment by leofeyer** [comment]
As discussed in Mumble on July 30th, we do not want to merge this PR.

File: manager-bundle/bin/contao-api (hunk 1 above)
Comment (discordier) [reply manager-bundle/bin/contao-api:35]: `unset()` triggers direct gc in PHP - therefore you also should pass all variables when unsetting multiple variables.

**Pull request closed without merging**